string_decode.lua

```diff
@@ -2,42 +2,21 @@
 -- String decoder functions
 --
 
-local stringdecode = {}
-
--- From http://lua-users.org/wiki/LuaUnicode
-local function utf8_to_32(utf8str)
-    assert(type(utf8str) == "string")
-    local res, seq, val = {}, 0, nil
-
-    for i = 1, #utf8str do
-        local c = string.byte(utf8str, i)
-        if seq == 0 then
-            table.insert(res, val)
-            seq = c < 0x80 and 1 or c < 0xE0 and 2 or c < 0xF0 and 3 or
-                  c < 0xF8 and 4 or --c < 0xFC and 5 or c < 0xFE and 6 or
-                error("Invalid UTF-8 character sequence")
-            val = bit32.band(c, 2^(8-seq) - 1)
-        else
-            val = bit32.bor(bit32.lshift(val, 6), bit32.band(c, 0x3F))
-        end
-
-        seq = seq - 1
-    end
+local lunicode = require(".unicode")
```

Comment:
Congratulations!
Yet the dot here must be a typo?

Reply:
Similar uses utf8.lua for relative require, and the unicode module is relative to string_decode.lua in the same directory currently, so it seemed appropriate
https://github.com/Stepets/utf8.lua/blame/master/README.md#L54

currently works even with tests, so I'm not sure it's wrong, however that might change with implementing #11